src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -1859,23 +1870,10 @@ public override void List(CriteriaImpl criteria, IList results)
 
 				string[] implementors = Factory.GetImplementors(criteria.EntityOrClassName);
 				int size = implementors.Length;
-
```

Comment:
Hi!
Unfortunately now I don't remember why I didn't refactor everything. Most probably I was hoping that my pull request can be included in current version of NH and I tried to minimize amount of (API) changes.
But since this pull request will make only in NH 5, I think, I could make a few more adjustments.